src/beast/beast/asio/streambuf.h

```diff
@@ -442,6 +442,7 @@ basic_streambuf<Allocator>::prepare (size_type n) ->
                     ++pos;
                     break;
                 }
+                out_end_ = pos->size();
```

Comment:
I'm open to an alternate implementation proposal, or proposed wording for a comment. Are you sure that the only thing changed is "when consuming the final block exactly satisfies the prepare request?"